### PR TITLE
[plugin.audio.soundcloud] 4.0.0

### PR DESCRIPTION
**Pull request by jaylinski**
### Description

This is an update for Matrix and based on https://github.com/xbmc/repo-plugins/tree/krypton/plugin.audio.soundcloud.

#### Changes

* Removed Python 2 code
* Updated settings to new version
* Switched to new `with`-syntax for file-operations (https://github.com/xbmc/xbmc/commit/0775d520a166e07322ffdd092d5c8e2758b1ddc2)

A diff can be viewed here: https://github.com/jaylinski/kodi-addon-soundcloud/pull/49

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.
- If you see no activity on your PR after a week (so at least one weekend has passed) then please go to the #kodi-dev freenode IRC channel to reach out to the team

### CODE REVIEW

File: plugin.audio.soundcloud/resources/settings.xml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+                            <option label="30064">1</option>
+                        </options>
+                    </constraints>
+                    <control type="spinner" format="string"/>
+                </setting>
+            </group>
+            <group id="4" label="30070">
+                <setting id="cache.clear" type="action" label="30071" help="">
+                    <level>1</level>
+                    <data>RunPlugin(plugin://plugin.audio.soundcloud/settings/cache/clear/)</data>
```
Comment (jaylinski) [inline plugin.audio.soundcloud/resources/settings.xml:80]: @ronie I noticed I can't use `$ID` anymore. Is this a bug or was that a deliberate decision?

See https://github.com/jaylinski/kodi-addon-soundcloud/blob/v3.2.0/resources/settings.xml#L15.

Should probably be mentioned in https://kodi.wiki/view/Add-on_settings_conversion.

Comment (ronie) [reply plugin.audio.soundcloud/resources/settings.xml:80]: i don't know i'm afraid... the wiki page for the 'old' addon settings does not mention $ID either.
maybe @Montellese can tell.

Comment (jaylinski) [reply plugin.audio.soundcloud/resources/settings.xml:80]: @ronie I opened a PR regarding the `$ID` replacement: https://github.com/xbmc/xbmc/pull/18959

This addon-PR can be merged in the meantime, if you approve it. :)

File: plugin.audio.soundcloud/resources/plugin.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+import urllib.parse
+import xbmc
+import xbmcaddon
+import xbmcgui
+import xbmcplugin
+
+addon = xbmcaddon.Addon()
+addon_id = addon.getAddonInfo("id")
+addon_base = "plugin://" + addon_id
+addon_profile_path = xbmc.translatePath(addon.getAddonInfo("profile"))
```
Comment (enen92) [inline plugin.audio.soundcloud/resources/plugin.py]: xbmcvfs.translatePath for Matrix and aboce

**Review by enen92 — CHANGES_REQUESTED** [review]
Just one change needed

File: plugin.audio.soundcloud/resources/plugin.py (hunk 2 above)
Comment (jaylinski) [reply plugin.audio.soundcloud/resources/plugin.py]: Thanks, fixed it.

**Review by enen92 — APPROVED** [review]

**Pull request merged**